dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ResolveLLVMObjectSizeCalls.java

```diff
@@ -0,0 +1,61 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.expression.ExpressionFactory;
+import com.dat3m.dartagnan.expression.IValue;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.functions.ValueFunctionCall;
+import org.sosy_lab.common.configuration.Configuration;
+
+import java.math.BigInteger;
+
+/*
+    https://llvm.org/docs/LangRef.html#llvm-objectsize-intrinsic
+    A call to llvm.objectsize(ptr, ...) returns the size of the (accessible) object pointed to by <ptr>.
+    For example, if ptr_base is returned by alloc(100), then llvm.objectsize(ptr_base) == 100,
+    but llvm.objectsize(ptr_base + 80) == 20 (~ there are only 20 accessible bytes at that offset).
+    If the size cannot be determined statically, the call is replaced by either 0 or -1.
+    The intrinsic is typically used together with bound-checked memset/memcpy calls (e.g. __memset_chk) where the bound
+    check may get statically eliminated.
+
+    NOTE: For simplicity, we will always assume the object size to be statically unknown and return -1/0.
+
+    FIXME: A correct implementation needs to traverse the use-def chain from a llvm.objectsize call back to
+     the allocation site while tracking offset computations.
+     Unfortunately, this is quite hard to do in the presence of array indexing which can result in dynamic offsets.
+     A way around this is to rely on unrolling + CP to resolve dynamic offsets, then run this pass, and finally
+     run a second CP to propagate the resolved object sizes.
+ */
+public class ResolveLLVMObjectSizeCalls implements FunctionProcessor {
```

Comment:
We should at least log a warning

Reply:
I don't think a warning here would be helpful because there is not a single case (for now) where this handling is unsound. After all, it is valid for LLVM to reduce the call to 0/-1 according to the specification.
I would rather add a warning to `__memset_chk` that shows we skipped a bound check.

Reply:
You are right ... The warning there makes more sense